exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -251,6 +252,11 @@ public void run() {
     final String originalName = currentThread.getName();
     currentThread.setName(queryIdString + ":foreman");
 
+    try {
+      checkForemanState();
+    } catch (ForemanException e) {
+      addToEventQueue(QueryState.FAILED, new ForemanException("Query submission failed since foreman is shutting down"));
```

Comment:
why are you not just adding 'e' to the eventQueue ? why do you have to create a new ForemanException object?